sqlglot/parser.py

```diff
@@ -1969,10 +1969,35 @@ def _parse_select(
 
             self._match_r_paren()
 
-            # early return so that subquery unions aren't parsed again
-            # SELECT * FROM (SELECT 1) UNION ALL SELECT 1
-            # Union ALL should be a property of the top select node, not the subquery
-            return self._parse_subquery(this, parse_alias=parse_subquery_alias)
+            alias = None
+
+            # Parse a wrapped table as a Paren, instead of a Subquery. The exception to this
+            # is when there's an alias that can be applied to the parentheses, because that
+            # would shadow all wrapped table names, and so we want to parse it as a Subquery
+            # to represent the inner scope appropriately. Additionally, we want the node under
+            # the Subquery to be an actual query, so we will replace the table reference with
+            # a star query that selects from it.
+            if isinstance(this, (exp.Table, exp.Paren)):
+                # Removes redundant parentheses, e.g. in "SELECT * FROM ((tbl1 JOIN tbl2)) AS t"
+                this = t.cast(exp.Expression, this.unnest())
+
+                alias = self._parse_table_alias()
+                if not alias:
+                    return exp.paren(this, copy=False)
+
+                joins = this.args.get("joins")
+                this.set("joins", None)
+
+                this = this.replace(exp.select("*").from_(this.copy()))
```

Comment:
from_ copy = False ? why do you need to call this.copy

Reply:
> why do you need to call this.copy

Because otherwise we get an infinite loop.

Without the copy you first construct an AST where `this`'s parent becomes the `FROM` and then you try to replace `this` with the produced `Select` expression, which contains a reference to it.